tegra-state-scripts: remove the correct file

focusware-develop opened the pull request:
When updating the nv_boot_control.conf file during the payload update the wrong file was removed. The "${mnt}/etc/nv_boot_control.conf" file is a symbolic link to "${mnt}/var/lib/nvbootctrl/nv_boot_control.conf". When the symbolic link in /etc is removed, the next firmware update will fail.

Instead of removing the symbolic link, the file it points to is removed and recreated after the setup-nv-boot-control script is executed.

Signed-off-by: Boris Nagels <develop@focusware.nl>

Commit:
tegra-state-scripts: remove the correct file

When updating the nv_boot_control.conf file during the payload update the wrong file was removed. The "${mnt}/etc/nv_boot_control.conf" file is a symbolic link to "${mnt}/var/lib/nvbootctrl/nv_boot_control.conf". When the symbolic link in /etc is removed, the next firmware update will fail.

Instead of removing the symbolic link, the file it points to is removed and recreated after the setup-nv-boot-control script is executed.

Signed-off-by: Boris Nagels <develop@focusware.nl>

madisongh commented:
Actually, removing the symlink in `/etc` **is** the intended behavior.  The format/contents of the `nv_boot_control.conf` file can change from release to release of the L4T BSP, so there's no point in storing the config file in the persistent `/var/lib` .  The `setup-nv-boot-control` script is invoked immediately after the removal so a correctly-formatted (for the new rootfs) copy of the conf file should get created.

The idea is to keep the conf file specific to each installed rootfs, since its contents is closely coupled to the NVIDIA tools in the rootfs.

madisongh committed:
tegra-mender-setup.bbclass: make parseable for non-tegra MACHINEs

This class really should only be inherited when building for
tegra platforms, but don't cause parsing failures if it
happens to be inherited for others.

Signed-off-by: Matt Madison <matt@madison.systems>
